app/web/Dockerfile

```diff
@@ -6,6 +6,19 @@ WORKDIR /app
 # disable next.js telemetry
 ENV NEXT_TELEMETRY_DISABLED=1
 
+COPY package.json yarn.lock ./
+# https://github.com/yarnpkg/yarn/issues/8242
+RUN yarn config set network-timeout 300000 && yarn install --frozen-lockfile
+
+COPY . .
+
+RUN cp .env.$environment env && \
+    rm .env.* && \
+    mv env .env.local
+
+RUN yarn build
+
+# These are at the end so as to not invalidate Docker build cache
```

Comment:
This looks a bit weird, but all I did was to move args/envs to the end, and merge a couple of RUNs